lib/matplotlib/collections.py

```diff
@@ -415,6 +415,17 @@ def draw(self, renderer):
         renderer.close_group(self.__class__.__name__)
         self.stale = False
 
+    def set_picker(self, p):
+        # docstring inherited
+        # After deprecation, the whole method can be deleted and inherited.
+        if isinstance(p, Number) and not isinstance(p, bool):
+            cbook.warn_deprecated(
+                "3.4", message="Setting the collections's pick radius via "
+                "set_picker is deprecated since %(since)s and will be removed "
+                "%(removal)s; use set_pickradius instead.")
+            self.set_pickradius(p)
```

Comment:
Should you then set `p` to `True` since you've removed the type checks later?

Reply:
and probably make the same change in `lines.py` as well?